agent/src/main/scala/za/co/absa/atum/agent/dispatcher/HttpDispatcher.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.atum.agent.dispatcher
+
+import com.typesafe.config.Config
+import org.apache.spark.internal.Logging
+import za.co.absa.atum.agent.model.MeasureResult
+import sttp.client3._
+import sttp.model.Uri
+import za.co.absa.atum.agent.AtumContext
+
+class HttpDispatcher(config: Config) extends Dispatcher with Logging {
+
+  private val serverUri = Uri.unsafeParse(config.getString("url"))
+  private val backend = HttpURLConnectionBackend()
+
+  logInfo("using http dispatcher")
+  logInfo(s"serverUri $serverUri")
+
+  override def publish(checkpointKey: String, measureResult: MeasureResult): Unit = {
+    basicRequest
+      .body(s"$checkpointKey $measureResult")
+      .post(serverUri)
+      .send(backend)
```

Comment:
How would retries work? I never used `sttp` library but I quickly checked it and it looks awesome!

Reply:
There is no direct retry support, but it could be implemented using: https://github.com/softwaremill/retry or https://github.com/hipjim/scala-retry